nsxt/resource_nsxt_policy_vpc_gateway_policy.go

```diff
@@ -0,0 +1,163 @@
+/* Copyright © 2024 Broadcom, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/bindings"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/data"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+
+	"github.com/vmware/terraform-provider-nsxt/api/infra/domains"
+	utl "github.com/vmware/terraform-provider-nsxt/api/utl"
+)
+
+func resourceNsxtPolicyVPCGatewayPolicy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtPolicyVPCGatewayPolicyCreate,
+		Read:   resourceNsxtPolicyVPCGatewayPolicyRead,
+		Update: resourceNsxtPolicyVPCGatewayPolicyUpdate,
+		Delete: resourceNsxtPolicyVPCGatewayPolicyDelete,
+		Importer: &schema.ResourceImporter{
+			State: nsxtPolicyPathResourceImporter,
+		},
+
+		Schema: getPolicyGatewayPolicySchema(false),
+	}
+}
+
+func resourceNsxtPolicyVPCGatewayPolicyCreate(d *schema.ResourceData, m interface{}) error {
+	connector := getPolicyConnector(m)
+
+	// Initialize resource Id and verify this ID is not yet used
```

Comment:
If `domain` is the only difference between this function and `resourceNsxtPolicyGatewayPolicyCreate`, would it make sense to parametrize this single value here and below?

Reply:
Both funcs barely have any logic. I don't see much value in adding excess of functions which in the end, do nothing, pretty much like in `Delete()`